account_financial_report/models/account_move_line.py

```diff
@@ -12,6 +12,16 @@ class AccountMoveLine(models.Model):
         "account.analytic.account", compute="_compute_analytic_account_ids", store=True
     )
 
+    analytic_plan_ids = fields.Many2many(
+        "account.analytic.plan", compute="_compute_analytic_plan_ids", store=True
+    )
+
+    @api.depends("analytic_account_ids")
+    def _compute_analytic_plan_ids(self):
+        for line in self:
+            line.analytic_plan_ids = False
+            line.analytic_plan_ids = self.analytic_account_ids.plan_id
```

Comment:
You're still assigning all the plans to all the lines. And as the computed field is stored you don't really need to init the values
```suggestion
    def _compute_analytic_plan_ids(self):
        for line in self:
            line.analytic_plan_ids = line.analytic_account_ids.plan_id
```